src/ReactTestWrapper.js

```diff
@@ -90,4 +90,8 @@ export default class ReactTestWrapper extends TestWrapper {
   hasRef (ref) {
     return !!this.wrapper.instance().refs[ref]
   }
+
+  type () {
+    throw new Error('full DOM rendering does not support `type` yet')
+  }
```

Comment:
That's a good idea.  I'll take a look at that when I have some more free time.  I agree that it would be strange for this assertion to only work with `shallow()`, so I was hoping someone would have an idea that would make it work for both.